frontend/src/components/utils/Authentication.tsx

```diff
@@ -0,0 +1,59 @@
+import axios from 'axios';
+import { useEffect } from 'react';
+import { useLocation, useHistory } from 'react-router-dom';
+
+const authenticatedRouts = ['/home'];
+const unauthenticatedRouts = ['/login', '/registration'];
+
+const Authentication = ({children}: any) => {
+    const history = useHistory();
+    const location = useLocation();
+
+    useEffect(() => {
+        let isAuthenticated = false;
+
+        const getUserRoute = () => {
+            let currentRoute;
+            let defaultRoute = '/login';
+
+            if(isAuthenticated) {
+                defaultRoute = '/home';
+                currentRoute = authenticatedRouts.find(route => route === location.pathname);
+            } else {
+                defaultRoute = '/login';
+                currentRoute = unauthenticatedRouts.find(route => route === location.pathname);
+            }
+
+            if(currentRoute) {
+                history.push(currentRoute);
+            } else {
+                history.push(defaultRoute);
+            }
+        }
+
+        try {
+            if(!localStorage.getItem('token')) {
+                isAuthenticated = false;
+                getUserRoute();
+            } else {
+                axios.get('http://localhost:3000/users').then(res => {
```

Comment:
This is also a security issue here, you are returning user password along with other less severe but nonetheless not so important fields.
![image](https://user-images.githubusercontent.com/5684735/115410644-dcec3000-a203-11eb-8871-659f97e00de6.png)